packages/playwright-core/src/server/injected/injectedScript.ts

```diff
@@ -1058,6 +1063,18 @@ export class InjectedScript {
       return { received, matches };
     }
 
+    if (expression === 'to.contain.class.array') {
+      const receivedClassNames = elements.map(e => e.className);
+
+      if (!Array.isArray(options.expectedValue) || options.expectedValue.length !== elements.length)
```

Comment:
This does not match the logic of `to.contain.text.array`, toContainClass relative to toHaveClass should be the same as toContainText relative to toHaveText, let's make sure they are consistent.

Reply:
Could you please give me a testcase for what you are saying? This condition just ensures that elements and values length are equal.